mshadow/cuda/tensor_gpu-inl.cuh

```diff
@@ -256,6 +256,27 @@ __global__ void SoftmaxGradKernel(DstPlan dst, SrcPlan1 src, SrcPlan2 label, ind
 }
 
 template<int x_bits, typename DType, typename DstPlan, typename SrcPlan1, typename SrcPlan2>
+__global__ void SmoothSoftmaxGradKernel(DstPlan dst, SrcPlan1 src, SrcPlan2 label, index_t xmax,
+                                        float alpha) {
+  const unsigned x_size = 1 << x_bits;
+  const int y = blockIdx.x;
+  const int k = static_cast<int>(label.Eval(0, y));
+  const float smooth_grad = (alpha / (xmax - 1));
```

Comment:
is `xmax` the number of classes?

Reply:
Yep.  number of classes / vocab size.

Reply:
I would add a comment

Reply:
Will do.  

The copy and paste here is a little too much imo.  I was thinking about refactoring and at the same time making the ignore_label argument a index_t.  I'm leaning towards doing this in a separate PR though, what do you think?

Reply:
comment added.

Reply:
separate PR sounds fine.